python/taichi/lang/impl.py

```diff
@@ -105,17 +106,19 @@ def expr_init_func(
 
 
 def begin_frontend_struct_for(ast_builder, group, loop_range):
-    if not isinstance(loop_range, (AnyArray, Field, SNode, _Root)):
+    if not isinstance(loop_range,
+                      (AnyArray, Field, SNode, RWTextureAccessor, _Root)):
         raise TypeError(
             f"Cannot loop over the object {type(loop_range)} in Taichi scope. Only Taichi fields (via template) or dense arrays (via types.ndarray) are supported."
         )
-    if group.size() != len(loop_range.shape):
+    if (not isinstance(loop_range, RWTextureAccessor)) and group.size() != len(
```

Comment:
Why not check shape for `RWTextureAccessor`?

Reply:
Let's also add a test for the invalid case :-)

Reply:
Done. :)))